File: Courses/PHP Web Basics - май 2019/14.Exercise_Defining Classes/15.CarSalesman/CarSalesman/Models/Car.php
<?php

namespace CarSalesman\Models;

class Car
{
    //Полета
   private $model;
   private $engine;
   private $weight;
   private $color;

   //Конструктор
    /**
     * Car constructor.
     * @param $model
     * @param $engine
     * @param $weight
     * @param $color
     */
    public function __construct(string $model, Engine $engine, int $weight = null, string $color = null)
    {
        $this->model = $model;
        $this->engine = $engine;
        $this->weight = $weight;
        $this->color = $color;
    }

    public function __toString(): string
    {
        $output = "{$this->model}:" . PHP_EOL;
        $output .= $this->engine;
        $weight = $this->weight == null ? "n/a" : $this->weight;
        $output .= "  Weight: {$weight}" . PHP_EOL;
        $color = $this->color == null ? "n/a" : $this->color;
        $output .= "  Color: {$color}";
        return $output;
    }

}

File: Courses/C# OOP/04. Defining Classes Exercises/DefiningClasses Exer/10. Car Salesman/Car.cs
using System.Text;

namespace CarSalesman
{
    public class Car
    {
        private string model;
        private Engine engine;
        private double? weight;
        private string color;

        public Car(string model, Engine engine)
        {
            Model = model;
            Engine = engine;
        }

        public string Model { get => model; set => model = value; }
        public Engine Engine { get => engine; set => engine = value; }
        public double? Weight { get => weight; set => weight = value; }
        public string Color { get => color; set => color = value; }


        public override string ToString()
        {
            StringBuilder builder = new StringBuilder();

            builder.AppendLine($"{Model}:");
            builder.Append(Engine);
            builder.AppendLine($"  Weight: {(Weight == null ? "n/a" : Weight.ToString())}");
            builder.Append($"  Color: {(Color == null ? "n/a" : Color)}");

            return builder.ToString();
        }
    }
}
